### PR TITLE
Add GET endpoint to retrieve payments with pending refund status

### PR DESCRIPTION
**Pull request by dalebradley**
Create a GET endpoint that gets all payments with refund-pending status as well as a total counter. 

BI-10293

### Type of change

* [ ] Bug fix
* [ x] New feature
* [ ] Breaking change

### Pull request checklist

* [ x] I have added unit tests for new code that I have added
* [ ] I have added/updated functional tests where appropriate
* [ ] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/go.md)

__An exhaustive list of peer review checks can be read [here](https://github.com/golang/go/wiki/CodeReviewComments)__

### CODE REVIEW

File: handlers/bulk_refunds.go [hunk 1]
```diff
@@ -124,3 +125,26 @@ func closeFile(file multipart.File) {
 		log.Error(fmt.Errorf("error closing file: %w", err))
 	}
 }
+
+// HandleGetRefundStatuses retreives payments that are pending refund
```
Comment (Joseph-Nash) [inline handlers/bulk_refunds.go:129]: *retrieves

File: handlers/register.go [hunk 2]
```diff
@@ -96,6 +96,7 @@ func Register(mainRouter *mux.Router, cfg config.Config, paymentsDao dao.DAO) {
 	// Admin router will handle all the routes with an admin prefix
 	// and will be intercepted to check for the admin role
 	adminRouter := mainRouter.PathPrefix("/admin/payments/bulk-refunds").Subrouter()
+	adminRouter.HandleFunc("", HandleGetRefundStatuses).Methods("GET").Name("get-refund-statuses")
```
Comment (Joseph-Nash) [inline handlers/register.go:99]: Need to add relevant test in register_test.go

File: interceptors/payment_authentication.go [hunk 3]
```diff
@@ -188,7 +187,7 @@ func PaymentAdminAuthenticationIntercept(next http.Handler) http.Handler {
 		ctx := context.WithValue(r.Context(), helpers.ContextKeyUserID, userEmail)
 
 		// Check that user has bulk refund role and sends POST request
```
Comment (rsachdevch) [inline interceptors/payment_authentication.go:189]: We should edit the comment to reflect that we're not checking for a POST request anymore

File: models/payment_rest.go [hunk 4]
```diff
@@ -78,3 +78,8 @@ type CostResourceRest struct {
 	ProductType             string            `json:"product_type"              validate:"required"`
 	DescriptionValues       map[string]string `json:"description_values"`
 }
+
+type PendingRefundPaymentsResourceRest struct {
```
Comment (rsachdevch) [inline models/payment_rest.go:82]: Missing comment on public struct

File: handlers/bulk_refunds.go [hunk 5]
```diff
@@ -124,3 +125,26 @@ func closeFile(file multipart.File) {
...
+func HandleGetRefundStatuses(w http.ResponseWriter, req *http.Request) {
```
Comment (rsachdevch) [inline handlers/bulk_refunds.go:130]: We should add unit tests for this method because Sonar has failed the quality checks due to missing tests

**Review by rsachdevch — APPROVED** [review]

**Pull request merged**